src/bluesky/run_engine.py

```diff
@@ -2357,6 +2358,9 @@ async def _wait(self, msg):
                     # bar.
                     self._call_waiting_hook(status_objs)
                 await self._wait_for(Msg("wait_for", None, futs, timeout=msg.kwargs.get("timeout", None)))
+                # Monitor the status objects for any failures and raise an exception if any are detected.
+                self._watch_status_objs(watch)
```

Comment:
Doesn't this mean that we don't start watching these groups until after the wait for is complete?